SonarQube.TeamBuild.PostProcessor/SummaryReportBuilder.cs

```diff
@@ -131,12 +134,43 @@ private void GenerateReports()
         private static string GetSonarDashboadUrl(AnalysisConfig config)
         {
             string hostUrl = config.SonarQubeHostUrl.TrimEnd('/');
+            string branch = GetBranch(config);
+
+            string sonarUrl;
+
+            if (String.IsNullOrWhiteSpace(branch))
+            {
+                sonarUrl = string.Format(
+                    System.Globalization.CultureInfo.InvariantCulture,
+                    DashboardUrlFormat,
+                    hostUrl,
+                    config.SonarProjectKey);
+            }
+            else
+            {
+                sonarUrl = string.Format(
+                   System.Globalization.CultureInfo.InvariantCulture,
+                   DashboardUrlFormatWithBranch,
+                   hostUrl,
+                   config.SonarProjectKey,
+                   branch);
+            }
 
-            string sonarUrl = string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                DashboardUrlFormat, hostUrl, config.SonarProjectKey);
             return sonarUrl;
         }
 
+        private static string GetBranch(AnalysisConfig config)
```

Comment:
TryGetBranch would be more consistent for a method that might return null. Or FindBranch.